src/main/java/com/treasure/hunt/strategy/searcher/impl/strategyFromPaper/BadHintSubroutine.java

```diff
@@ -0,0 +1,227 @@
+package com.treasure.hunt.strategy.searcher.impl.strategyFromPaper;
+
+import com.treasure.hunt.strategy.hint.impl.HalfPlaneHint;
+import com.treasure.hunt.strategy.searcher.Movement;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.LineSegment;
+import org.locationtech.jts.geom.Point;
+
+import static com.treasure.hunt.strategy.hint.impl.HalfPlaneHint.Direction.*;
+import static com.treasure.hunt.strategy.searcher.impl.strategyFromPaper.GeometricUtils.*;
+import static com.treasure.hunt.strategy.searcher.impl.strategyFromPaper.RoutinesFromPaper.*;
+import static com.treasure.hunt.utils.JTSUtils.GEOMETRY_FACTORY;
+import static com.treasure.hunt.utils.JTSUtils.lineWayIntersection;
+import static org.locationtech.jts.algorithm.Angle.normalizePositive;
+
+/**
+ * @author bsen
+ */
+public class BadHintSubroutine {
+
+    /**
+     * If the last hint was bad, this function can be called and lastBadHint has to be set accordingly.
+     * The function equals the "else"-part of the first if-condition in Algorithm 3 (Function ReduceRectangle(R))
+     * in the paper.
+     * Variable names are equivalent to the paper, but since a', d', etc. is not a valid variable name in Java,
+     * _apos is used in such cases (apos for apostrophe), e.g. a_apos in this implementation equates to a'
+     * in the paper. _doubleApos signals a double apostrophe.
+     * At, Bt, Ct and Dt in this implementation equate to A, B, C and D in the paper, since the
+     * not by phi transformed variables of the current rectangle R are also stored with A, B, C and D.
+     * The t signals the transformed state of these variables.
+     * hintT equates to the hint (L1', x1')
+     *
+     * @param curHint
+     * @return The move to scan various areas so that A,B,C and D can be updated to a smaller rectangle (or the treasure
+     * is found)
+     */
+    static Movement lastHintBadSubroutine(StrategyFromPaper strategy, HalfPlaneHint curHint, HalfPlaneHint lastBadHint,
```

Comment:
Why is everything static , maybe make it a singleton or well u can also leave it